### ISSUE TITLE
Reproducing results on the Segmentation task

### ISSUE THREAD

**Issue by Ankowa** [facebookresearch/dinov2#385]
Has anyone been able to reproduce the results from the paper for the segmentation task in the linear probing scenario? I've been trying to get decent results for the ADE20k and CityScapes datasets, and the best I could get for the ViT-Small/14 model was 0.33 mIoU on the former (vs 0.44 in the paper), and 0.58 for the latter (vs 0.67).

The training setup is as follows:
* batch size 32
* lr of 0.0005
* 100 epochs (starts to overfit around 30th epoch, the best eval mIoU is reported)
* normalization of the input based on dataset-specific means and stds
* augmentation: random horizontal flip (from albumentations library, no bugs here)
* input to the linear layer: patch embeddings of the last attention head, the output is logits per patch, which is then interpolated (using linear interpolation) to the input size.

The implementation was roughly based on [this tutorial](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/DINOv2/Train_a_linear_classifier_on_top_of_DINOv2_for_semantic_segmentation.ipynb), and is in pure pytorch (no mmseg used).

Forward pass is implemented as follows:
```
    def forward(self, embeddings: torch.Tensor) -> torch.Tensor:
        embeddings = embeddings.reshape(-1, self.height, self.width, self.in_channels)
        embeddings = embeddings.permute(0, 3, 1, 2)

        logits = self.classifier(embeddings)
        logits = torch.nn.functional.interpolate(
            logits, size=self.output_size, mode="bilinear", align_corners=False
        )

        return logits
```

Any tips on that?

**Comment by Ankowa** [comment]
Update on this based on [config files from notebook](https://dl.fbaipublicfiles.com/dinov2/dinov2_vits14/dinov2_vits14_voc2012_linear_config.py):
* batch size 16, lr 0.001
* 50 epochs (~40k iters)
* normalization based on imagenet means and stds
* augmentations adjusted from the mmseg config above
* correct implementation of mIoU evaluation function (the one from huggingface.evaluate does not ignore the background class, issue in progress)
* added batchnorm before conv layer

The results on all encoders closely match (up to 0.03 drop in mIoU), I am satisified.

**Comment by franchesoni** [comment]
Hi @Ankowa did you end up using mmseg or your custom training code?

**Comment by Ankowa** [comment]
@franchesoni I've used pure pytorch. I can't share code now, unfortunately, but what I've done was basically to implement all the transformations in the config files, as well as the sliding window inference for evaluation. It took a while to get it right, but yields comparable results (at most 0.03 mIoU drop compared to results from the paper for dinov2). I'll link code once I am able to (I can't promise it will be still relevant once it's released, apologies)

**Comment by JianwenCao** [comment]
@Ankowa
What did you do with the test preprocessor, I followed the document you posted, but the miou drops a lot. I did not even train my own model, I just downloaded the provided model head.

**Comment by manugaurdl** [comment]
> [@franchesoni](https://github.com/franchesoni) I've used pure pytorch. I can't share code now, unfortunately, but what I've done was basically to implement all the transformations in the config files, as well as the sliding window inference for evaluation. It took a while to get it right, but yields comparable results (at most 0.03 mIoU drop compared to results from the paper for dinov2). I'll link code once I am able to (I can't promise it will be still relevant once it's released, apologies)

Hey, can you release the code now? Will help a lot :)
